packages/stream/components/utils.ts

```diff
@@ -0,0 +1,32 @@
+type PendingThenableState = { status: "pending" };
+type FulfilledThenableState<T> = { status: "fulfilled"; value: T };
+type RejectedThenableState = { status: "rejected"; reason: unknown };
+
+type ThenableState<T> =
+	| PendingThenableState
+	| FulfilledThenableState<T>
+	| RejectedThenableState;
+
+export type Thenable<T> = Promise<T> & ThenableState<T>;
+
+export function trackThenableState<T>(promise: Promise<T>): Thenable<T> {
```

Comment:
```suggestion
export function trackPromiseState<T>(promise: Promise<T>): Thenable<T> {
```

Given this takes a promise and returns a thenable, I think calling it `trackPromiseState` would make more sense

Reply:
i'm good with `trackPromiseState`, renamed